server/main.lua

```diff
@@ -91,13 +91,26 @@ end)
 lib.callback.register('qbx_truckrobbery:server:giveReward', function(source)
 	if Entity(truck).state.truckstate ~= TruckState.LOOTABLE then return end
 	Entity(truck).state:set('truckstate', TruckState.LOOTED, true)
+    local cantCarryRewards = {}
+    local cantCarryRewardsSize = 0
     for i = 1, #config.rewards do
         local reward = config.rewards[i]
         if not reward.probability or math.random() <= reward.probability then
             local amount = math.random(reward.minAmount or 1, reward.maxAmount or 1)
-            exports.ox_inventory:AddItem(source, reward.item, amount)
+            if exports.ox_inventory:CanCarryItem(source, reward.item, amount) then
+                exports.ox_inventory:AddItem(source, reward.item, amount)
+            else
+                cantCarryRewards[cantCarryRewardsSize] = reward.item
+                cantCarryRewardsSize += 1
+                cantCarryRewards[cantCarryRewardsSize] = amount
+                cantCarryRewardsSize += 1
```

Comment:
Shouldnt it just be this?

```lua
cantCarryRewards[cantCarryRewardsSize] = {reward.item, amount}
```

Reply:
This is in a loop so it can add more than one reward.item, amount pair

Reply:
> This is in a loop so it can add more than one reward.item, amount pair

yeah obviously you would need to do the += 1 above it, but why are you doing it twice?

Reply:
Oh. I somehow got confused and thought we needed to flatten the array instead of an array of tables. Also forgot that we need to start the array at 1 instead of 0 lol. Fixed.